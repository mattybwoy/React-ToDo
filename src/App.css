body{
  background-color: rgb(44, 44, 48);
}
.App{
  text-align: center;
  background-color: rgba(0, 255, 149, 0.11);
  margin: 40px auto;
  width: 500px;
  min-height: 500px;
}

#todo-form input {
  border: 0;
  height: 20px;
  padding: 0 20px;;
  margin: 10px;
  font-size: 16px;
  border-radius: 5px;
}

#todo-form button{
  height: 20px;
  width: 80px;
  border-radius: 30px;;
  cursor: pointer;
  font-weight: bolder;
}

