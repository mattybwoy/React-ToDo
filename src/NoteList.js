import React, { Component } from "react";


class NoteList extends Component {
constructor(props) {
  super(props)
  this.state = { value: '', todo: [], checked: false }
  
this.handleChange = this.handleChange.bind(this);
this.handleSubmit = this.handleSubmit.bind(this);
}

 handleChange(event) {
   this.setState({value: event.target.value});
 }

 handleSubmit(event) {
   this.state.todo.push(this.state.value)
   this.setState({todo: this.state.todo});
   event.preventDefault();
 }

 handleCheckboxChange = event =>
   this.setState({checked: event.target.checked})
 
   resetForm = () => {this.setState({
       ...this.state,
       value: '',
       todo: []
     })
   }
  
render() {
  return (
    <div className = 'Personal NoteList'>
    <form onSubmit = {this.handleSubmit}>
      <label>
        Task: <input type='text' value={this.state.value}onChange = {this.handleChange}/>
      </label>
      <input type = 'submit' value = 'Add'/ >
      <input type='button' value='Reset' onClick ={this.resetForm}/>
      </form>

      {this.state.todo.map((todo, index) => {
        return (<p key ={index}>{todo}
        <input type='checkbox' />
        </p>)
        })
      }
      </div>
  )
}

}

export default NoteList