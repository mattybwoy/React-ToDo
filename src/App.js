import React from 'react';
import './App.css';
import NoteList from './NoteList'
import NoteListHook from './NoteListHook'

function App() {
  return (
    <div className="App">
        <h1>My To Do List</h1>
        {/* < NoteList/> */}
        < NoteListHook/>
    </div>
  );
}

export default App;
