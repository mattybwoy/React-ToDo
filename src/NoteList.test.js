import React from 'react';
import '@testing-library/jest-dom'
import { getAllByTitle, render,screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer';
import App from './App';
import NoteList from './NoteList'


test('renders correctly when there are no notes', () => {
  const tree = renderer.create(<NoteList />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Show remaining tasks', () => {
  render(<App /> );
  expect(screen.getByText('You have 0 Tasks Remaining')).toBeInTheDocument()
});

test('user can add a task', () => {
  render(<NoteList /> );
  const textBox = screen.getByRole('textbox');
  userEvent.type(textBox, 'A task')
  userEvent.click(screen.getByText('Add'))
  const note = screen.getByText('A task');

  expect(note).toBeInTheDocument();
});

test('user can delete a task', () => {
  render (<NoteList />);
  const textBox = screen.getByRole('textbox');
  userEvent.type(textBox, 'A Task')
  userEvent.click(screen.getByText('Add'))
  const note = screen.getAllByText('A Task')
  userEvent.click(screen.getByTitle('deletion'))

  expect(screen.getByText('You have 0 Tasks Remaining')).toBeInTheDocument()
})

test('user see checkbox next to todo', () => {
  render(<NoteList />);
  const textBox = screen.getByRole('textbox');
  userEvent.type(textBox, 'A task')
  userEvent.click(screen.getByText('Add'))
  const note = screen.getByText('A task');
  const checkbox = screen.getByRole('checkbox')

  expect(checkbox).toBeInTheDocument();
})

test('user can reset todo list', () => {
render(<NoteList />);
  const textBox = screen.getByRole('textbox');
  userEvent.type(textBox, "A task")
  userEvent.click(screen.getByText('Add'))
  const note = screen.getByText('A task');
  userEvent.click(screen.getByText('Reset'))

  expect(note).not.toBeInTheDocument();
});

