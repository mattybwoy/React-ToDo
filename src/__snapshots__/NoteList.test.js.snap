// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no notes 1`] = `
<div
  className="Personal NoteList"
>
  <form
    id="todo-form"
    onSubmit={[Function]}
  >
    <label>
      <input
        onChange={[Function]}
        placeholder="Enter Task"
        type="text"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Add
    </button>
    <button
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
  </form>
  You have 
  0
   Tasks Remaining
  <footer>
    <strong>
      A MATTYBWOY CREATION
    </strong>
    <br />
    Powered by React
    <br />
    <img
      alt="https://reactjs.org/"
      src="logo.svg"
      width="40"
    />
  </footer>
</div>
`;
